Integrations/integration-Jask_v2.yml

```diff
@@ -0,0 +1,1540 @@
+            'ContentsFormat': formats['json']
+        })
+
+
+    ''' FETCH INCIDENTS '''
+
+    def fetch_incidents():
+        """
+        Retrieve new incidents periodically based on pre-defined instance parameters
+        We use the seen list to determine if it's new or not.
+        """
+        now = datetime.now()
+
+        if FETCH_DAYS > 0:
+            fetch_from = date.today() - timedelta(days=FETCH_DAYS)
+            fetch_from = fetch_from.strftime("%Y-%m-%d")
+        else:
+            fetch_from = date.today().strftime("%Y-%m-%d")
+
+        last_run_object = demisto.getLastRun()
+        if 'seen_insights' in last_run_object.keys():
+            seen = last_run_object['seen_insights']
+            # demisto.info("We have a last run object, previous last_run: {}".format(last_run))
+        else:
+            seen = []
+            # demisto.info("No last run object, creating new last_run: {}".format(last_run))
+
+        # periodically cleanup the seen list, because if you have more than 2000 a day... you got issues.
+        if len(seen) > 2000:
+            seen = []
+
+        # go get some insights!
+        # override the query is a specific fetch query in jask query lang is provided
+        if demisto.getParam('fetchQuery'):
+            q = demisto.getParam('fetchQuery')
+        else:
+            q = '-status:"closed" created:>={}&sort=-timestamp'.format(fetch_from)
+
+        query = QUERY.copy()
+        query['q'] = q
+        query['offset'] = 0
+        query['limit'] = FETCH_LIMIT
+        resp_json = jask_api_request('GET', 'insights', query)
+
+        # determine if we've seen them before, and open new incidents if we haven't
+        incidents = []
+        for insight in resp_json['data']['objects']:
+            current_fetch = insight.get('id')
+            if current_fetch not in seen:
+                incidents.append({
+                    'name': insight.get('name', 'No name') + ' - ' + insight.get('id'),
+                    'occurred': insight.get('timestamp') + 'Z',
+                    'details': insight.get('description'),
+                    'severity': 2,
+                    'rawJSON': json.dumps(insight)
+                })
+                seen.append(current_fetch)
+
+        # set last run with our seen list and create incidents
+        demisto.setLastRun({'time': datetime.strftime(now, "%Y-%m-%dT%H:%M:%S"),
+                            'seen_insights': seen})
+        demisto.incidents(incidents)
+        demisto.info("Jask Fetch Log: Fetched - {}, New - {}, Query - {}, Seen List: {}".format(len(resp_json['data']['objects']), len(incidents), q, seen))
+
+
+    ''' EXECUTION '''
+    LOG('command is %s' % (demisto.command(), ))
+
+    try:
+        if demisto.command() == 'test-module':
+            # validate the apiKey by querying the jask users/current api
+            user = jask_api_request('GET', 'users/current')
+            if user['data']['apiKey'] == APIKEY:
```

Comment:
Please avoid accessing dictionary keys directly, as it might raise `KeyError`.
Use `.get()` instead.